fabric-biomes-v1/src/main/java/net/fabricmc/fabric/api/biomes/v1/OverworldBiomes.java

```diff
@@ -0,0 +1,104 @@
+	 * 	             Vanilla weights for each biome in each climate are listed in the {@link OverworldClimate} javadoc.
+	 */
+	public static void addBaseBiome(Biome biome, OverworldClimate climate, double weight) {
+		InternalBiomeData.addOverworldBaseBiome(climate, biome, weight);
+	}
+
+	/**
+	 * Adds the biome as a hills variant of the parent biome, with the specified weight
+	 *
+	 * @param parent the biome to where the hills variant is added
+	 * @param hills the biome to be set as a hills variant
+	 * @param weight the weight of the entry. The weight in this method corresponds to its selection likelihood, with
+	 *               heavier biomes being more likely to be selected and lighter biomes being selected with less likelihood.
+	 *               Mods should use 1.0 as the default/normal weight.
+	 */
+	public static void addHillsBiome(Biome parent, Biome hills, double weight) {
+		InternalBiomeData.addOverworldHillsBiome(parent, hills, weight);
+	}
+
+	/**
+	 * Adds the biome as a shore/beach biome for the parent biome, with the specified weight
+	 *
+	 * @param parent the base biome to where the shore biome is added
+	 * @param shore the biome to be added as a shore biome
+	 * @param weight the weight of the entry. The weight in this method corresponds to its selection likelihood, with
+	 * 	             heavier biomes being more likely to be selected and lighter biomes being selected with less likelihood.
+	 * 	             Mods should use 1.0 as the default/normal weight.
+	 */
+	public static void addShoreBiome(Biome parent, Biome shore, double weight) {
+		InternalBiomeData.addOverworldShoreBiome(parent, shore, weight);
+	}
+
+	/**
+	 * Adds the biome as an an edge biome (excluding as a beach) of the parent biome, with the specified weight
+	 *
+	 * @param parent the base biome to where the edge biome is added
+	 * @param edge the biome to be added as an edge biome
+	 * @param weight the weight of the entry. The weight in this method corresponds to its selection likelihood, with
+	 *               heavier biomes being more likely to be selected and lighter biomes being selected with less likelihood.
+	 *               Mods should use 1.0 as the default/normal weight.
+	 */
+	public static void addEdgeBiome(Biome parent, Biome edge, double weight) {
+		InternalBiomeData.addOverworldEdgeBiome(parent, edge, weight);
+	}
+
+	/**
+	 * Adds a 'variant' biome which replaces another biome on occasion.
+	 * For example, addBiomeVariant(Biomes.JUNGLE, Biomes.DESERT, 0.2) will replace 20% of jungles with deserts.
+	 *
+	 * @param replaced the base biome that is replaced by a variant
+	 * @param variant the biome to be added as a variant
+	 * @param chance the chance of replacement of the biome into the variant
+	 */
+	public static void addBiomeVariant(Biome replaced, Biome variant, double chance, OverworldClimate... climates) {
```

Comment:
I am assuming that if you want to add an ocean biome you would have to add it as a variant of the ocean biome using this method, which, if you don't know how the worldgen works, probably feels a bit unintuitive.
Maybe consider adding a wrapper method specifically for adding sea biomes?

Reply:
@coderbot16 I'll leave this up to you to decide

Reply:
I think that it would be better to add this to the documentation somehow instead of creating a separate special case that would have identical functionality. Could be added to addBiomeVariant, or even to OverworldBiomes?

Reply:
I agree with @coderbot16

It would be better to add this to the documentation

Also I don't believe you can specifically set things as a variant of temperature ocean biomes as they only get applied near the end

Reply:
There is an AddMushroomIslandLayer if someone wants to make a mixin to that

Reply:
Documentation added in latest merge.